File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/mobilePush/RemoveDeviceFromChannelOthers.java
package com.pubnub.docs.mobilePush;

import com.pubnub.api.PubNubException;
import com.pubnub.api.enums.PNPushType;
import com.pubnub.api.java.PubNub;
import com.pubnub.docs.SnippetBase;

import java.util.Arrays;

public class RemoveDeviceFromChannelOthers extends SnippetBase {
    private void removeDeviceFromChannel() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/mobile-push#remove-device-from-channel-1
        // removePushNotificationsFromChannels

        PubNub pubNub = createPubNub();

        // snippet.removeDeviceFromChannel
        pubNub.removePushNotificationsFromChannels()
                .deviceId("googleDevice")
                .channels(Arrays.asList("ch1", "ch2", "ch3"))
                .pushType(PNPushType.FCM)
                .async(result -> { /* check result */ });
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/mobilePush/RemoveDeviceFromChannelOthers.kt
package com.pubnub.docs.mobilePush

import com.pubnub.api.enums.PNPushEnvironment
import com.pubnub.api.enums.PNPushType
import com.pubnub.docs.SnippetBase

class RemoveDeviceFromChannelOthers : SnippetBase() {
    private fun removePushNotificationsFromChannels() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/mobile-push#remove-device-from-channel-1
        val pubnub = createPubNub()

        // snippet.removePushNotificationsFromChannels
        // for FCM/GCM
        pubnub.removePushNotificationsFromChannels(
            deviceId = "googleDevice",
            channels = listOf("ch1", "ch2", "ch3"),
            pushType = PNPushType.FCM
        ).async { result -> }

        // for APNS2
        pubnub.removePushNotificationsFromChannels(
            deviceId = "appleDevice",
            channels = listOf("ch1", "ch2", "ch3"),
            pushType = PNPushType.APNS2,
            topic = "myapptopic",
            environment = PNPushEnvironment.DEVELOPMENT
        ).async { result -> }
        // snippet.end
    }
}
